src/commands/deployment/get.ts

```diff
@@ -151,8 +153,9 @@ export const getDeployments = (
 /**
  * Initializes the pipelines assuming that the configuration has been loaded
  */
-const initialize = () => {
+const initialize = async () => {
   const config = Config();
+  const key = await Config().introspection!.azure!.key;
```

Comment:
The `key` property implements a lazy `getter` that fetches the storage access key either from `Azure KeyVault` or `Config` file. It uses a `Promise` to avoid circular calls i.e. while loading `Config()` object first time, it makes a call to `keyVault` which requires `service_principal_id`, `service_principal_secret`, and `tenant_id` from the `Config()` object that goes into a circular loop. The `Promise` allows us to delay fetching the key value until any part of the code requires it. Does it makes sense?